plaso/filters/parser_filter.py

```diff
@@ -0,0 +1,201 @@
+        for parser_filter in excluded_parser_filters]
+    parser_filters.extend(included_parser_filters)
+    return ','.join(parser_filters)
+
+  def _ExpandPreset(self, preset_name, parsers_and_plugins):
+    """Expands a preset in the parsers and plugins.
+
+    This functions replaces the preset in parsers_and_plugins with the parser
+    and plugin or preset names defined by the preset.
+
+    Args:
+      preset_name (str): name of the peset to expand.
+      parsers_and_plugins (dict[str, set[str]]): parsers and plugins.
+
+    Raises:
+      RuntimeError: if the preset in parsers_and_plugins has plugins.
+    """
+    if preset_name not in parsers_and_plugins:
+      return
+
+    plugins = parsers_and_plugins.get(preset_name, None)
+    if plugins != set(['*']):
+      print(plugins)
+      raise RuntimeError('Unsupported preset: {0:s} with plugins.'.format(
+          preset_name))
+
+    del parsers_and_plugins[preset_name]
+
+    for parser_filter in self._presets_manager.GetParsersByPreset(preset_name):
+      parser, _, plugin = parser_filter.partition('/')
+      if not plugin:
+        plugin = '*'
+
+      parsers_and_plugins.setdefault(parser, set())
+      parsers_and_plugins[parser].add(plugin)
+
+  def _ExpandPresets(self, preset_names, parsers_and_plugins):
+    """Expands the presets in the parsers and plugins.
+
+    This functions replaces the presets in parsers_and_plugins with the parser
+    and plugin or preset names defined by the presets.
+
+    Args:
+      preset_names (set[str]): names of the presets defined by the presets
+          manager.
+      parsers_and_plugins (dict[str, set[str]]): parsers and plugins.
+    """
+    for preset_name in preset_names:
+      self._ExpandPreset(preset_name, parsers_and_plugins)
+
+  def ExpandPresets(self, expression):
+    """Expands all presets in a parser filter expression.
+
+    Args:
+      expression (str): the parser filter expression, where None
+          represents all parsers and plugins.
+
+    Returns:
+      str: a parser filter expression where presets have been expanded or None
+          to represent all parsers and plugins.
+    """
+    preset_names = set(self._presets_manager.GetNames())
+
+    excludes, includes = self.SplitExpression(expression)
+
+    while set(excludes.keys()).intersection(preset_names):
+      self._ExpandPresets(preset_names, excludes)
+
+    while set(includes.keys()).intersection(preset_names):
+      self._ExpandPresets(preset_names, includes)
+
+    return self._JoinExpression(excludes, includes)
+
+  def SplitExpression(self, expression):
+    """Determines the excluded and included parsers form an expression string.
+
+    Args:
+      expression (str): the parser filter expression, where None
+          represents all parsers and plugins.
+
+    Returns:
+      tuple: contains:
+
+        excludes (dict[str, set[str]]): excluded parsers and plugins.
+        includes (dict[str, set[str]]): included parsers and plugins.
```

Comment:
And a empty dictionary here also means all parsers and plugins, right?

Reply:
not in the context of this method